app/customer.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from app.car import Car
+from app.shop import Shop
+
+
+class Customer:
+    customers = {}
+
+    def __init__(
+            self,
+            name: str,
+            product_cart: dict,
+            location: list[int],
+            money: int | float,
+            car: Car
+    ) -> None:
+        self.__name = name
+        self.__product_cart = product_cart
+        self.__home_location = location
+        self.__current_location = self.__home_location
+        self.__money = money
+        self.__car = car
+        Customer.customers[self.__name] = self
+
+    @property
+    def name(self) -> str:
+        return self.__name
+
+    @property
+    def location(self) -> list[int]:
+        return self.__current_location
+
+    @location.setter
+    def location(self, new_location: list[int]) -> None:
+        self.__current_location = new_location
+
+    @property
+    def product_cart(self) -> dict:
+        return self.__product_cart
+
+    @property
+    def car(self) -> Car:
+        return self.__car
+
+    @property
+    def money(self) -> int | float:
+        return self.__money
+
+    @money.setter
+    def money(self, new_amount_money: int | float) -> None:
+        self.__money = new_amount_money
+
+    def shop_trip_cost(self, shop: Shop) -> float:
+        distance_to_shop = ((self.location[0] - shop.location[0]) ** 2
+                            + (self.location[1] - shop.location[1])
+                            ** 2) ** 0.5
+        products_cost = sum([
+            price * self.product_cart.get(product)
+            for product, price in shop.products_price.items()
+        ])
+        return round(
+            products_cost
+            + distance_to_shop * 2
+            * self.car.get_fuel_price_per_1_km, 2
+        )
+
+    def can_shop_here(self, shop: Shop) -> bool:

```

Comment:
not clearly naming. If the function returns bool var, function should start with `is_`